CONTRIBUTING.md

```diff
@@ -82,3 +90,55 @@ export class MyConstruct extends Construct {
   * Other import methods normally require access to the context API which is beyond the scope of these constructs. We'd like to address this in the future.
 * Constructs should implement an interface so that the import methods (e.g. `.fromSomethingAttributes()`) can return the interface type.
 
+
+
+## Testing
+
+Tests are fundamental to building good constructs. The following testing guidelines should be followed:
+
+* All tests should be stored in the `test/aws-<something>` subdirectory. For example, if you are creating
+a construct for the AppStream service, then the module name would be `aws-appstream` and tests would go in the `/test/aws-appstream` directory.
+* Fine-grained tests should be implemented for all constructs with a goal of 100% coverage. For example,
+if you were creating an Application construct in the AppSteam module, your tests should be in the `test/aws-appstream/application.test.ts` file.
+* Integration tests should be written for all constructs. For example, if you are creating an integration test
+for the Application construct in the AppStream module, your tests should be in the `test/aws-appstream/application.integ.ts` file.
+
+### Fine-grained assertion tests
+
+Unit tests should be created that cover all pathways of code in your construct, including the testing of any 
+default value or conditionals. Test reports will report code coverage. 
+
+For example, if your construct has a property that can be overridden, you should have a test that ensures the
+property is overridden correctly and that creates a default value when not overridden. 
+An example of this can be found [here](./test/aws-examplemodule/example.test.ts). 
+
+* In the test 'Uses provided Role for Lambda Function' the Lambda Function is referenced via the `.node.defaultChild` property and then the role is compared to the one provided.
```

Comment:
This approach could be brittle to refactorings. We could also test the same thing by verifying that the changed role in the template matches e.g. the name we set on the manually added role. What do you think?

Reply:
I've added a note about it, do you think that's sufficient?

Reply:
Thanks! Sorry for not answering sooner :)